conformance/tests/httproute-request-percentage-mirror.go

```diff
@@ -0,0 +1,236 @@
+	totalRequests         = 500.0
+	numDistributionChecks = 5
+)
+
+func init() {
+	ConformanceTests = append(ConformanceTests, HTTPRouteRequestPercentageMirror)
+}
+
+var HTTPRouteRequestPercentageMirror = suite.ConformanceTest{
+	ShortName:   "HTTPRouteRequestPercentageMirror",
+	Description: "An HTTPRoute with percentage based request mirroring",
+	Manifests:   []string{"tests/httproute-request-percentage-mirror.yaml"},
+	Features: []features.FeatureName{
+		features.SupportGateway,
+		features.SupportHTTPRoute,
+		// features.SupportHTTPRouteRequestPercentageMirror,
+	},
+	Test: func(t *testing.T, suite *suite.ConformanceTestSuite) {
+		var (
+			ns      = "gateway-conformance-infra"
+			routeNN = types.NamespacedName{Name: "request-percentage-mirror", Namespace: ns}
+			gwNN    = types.NamespacedName{Name: "same-namespace", Namespace: ns}
+			gwAddr  = kubernetes.GatewayAndHTTPRoutesMustBeAccepted(t, suite.Client, suite.TimeoutConfig, suite.ControllerName, kubernetes.NewGatewayRef(gwNN), routeNN)
+		)
+
+		kubernetes.HTTPRouteMustHaveResolvedRefsConditionsTrue(t, suite.Client, suite.TimeoutConfig, routeNN, gwNN)
+
+		testCases := []http.ExpectedResponse{
+			{
+				Request:   http.Request{Path: "/percent-mirror"},
+				Namespace: ns,
+				ExpectedRequest: &http.ExpectedRequest{
+					Request: http.Request{
+						Path: "/percent-mirror",
+					},
+				},
+				Backend: "infra-backend-v1",
+				MirroredTo: []http.MirroredBackend{
+					{
+						BackendRef: http.BackendRef{
+							Name:      "infra-backend-v2",
+							Namespace: ns,
+						},
+						Percent: ptrTo(int32(20)),
+					},
+				},
+			}, {
+				Request:   http.Request{Path: "/percent-mirror-fraction"},
+				Namespace: ns,
+				ExpectedRequest: &http.ExpectedRequest{
+					Request: http.Request{
+						Path: "/percent-mirror-fraction",
+					},
+				},
+				Backend: "infra-backend-v1",
+				MirroredTo: []http.MirroredBackend{
+					{
+						BackendRef: http.BackendRef{
+							Name:      "infra-backend-v2",
+							Namespace: ns,
+						},
+						Percent: ptrTo(int32(50)),
+					},
+				},
+			}, {
+				Request: http.Request{
+					Path: "/percent-mirror-and-modify-headers",
+					Headers: map[string]string{
+						"X-Header-Remove":     "remove-val",
+						"X-Header-Add-Append": "append-val-1",
+					},
+				},
+				ExpectedRequest: &http.ExpectedRequest{
+					Request: http.Request{
+						Path: "/percent-mirror-and-modify-headers",
+						Headers: map[string]string{
+							"X-Header-Add":        "header-val-1",
+							"X-Header-Add-Append": "append-val-1,header-val-2",
+							"X-Header-Set":        "set-overwrites-values",
+						},
+					},
+					AbsentHeaders: []string{"X-Header-Remove"},
+				},
+				Namespace: ns,
+				Backend:   "infra-backend-v1",
+				MirroredTo: []http.MirroredBackend{
+					{
+						BackendRef: http.BackendRef{
+							Name:      "infra-backend-v2",
+							Namespace: ns,
+						},
+						Percent: ptrTo(int32(35)),
+					},
+				},
+			},
```

Comment:
Why are we testing it jointly with the `requestHeaderModifier` filter? To assess that both filters are properly applied?

Reply:
exactly, thats a practice I always used to include in tests I developed. To make sure the feature is not messing up when applied with other standard feature. Does that make sense?

Reply:
yep, it makes sense to me 👍🏻